javaslang/src/test/java/javaslang/collection/IteratorTest.java

```diff
@@ -515,4 +517,33 @@ public void shouldRecognizeUnequalObjects() {
         // Iterator equality undefined
     }
 
+    // -- class initialization (see #1773)
+
+    @Test(timeout = 5_000)
+    public void shouldNotDeadlockOnConcurrentClassInitialization() throws InterruptedException {
+        final ExecutorService executorService = Executors.newFixedThreadPool(2);
+        executorService.execute(new ClassInitializer(javaslang.collection.Iterator.class.getName()));
```

Comment:
Iterator and AbstractIterator will be loaded in the main thread, thus, no concurrency.
You have to pass string constants here.